cider/tests/utils/CiderBatchChecker.cpp

```diff
@@ -251,6 +251,10 @@ bool CiderBatchChecker::checkOneStructBatchEqual(CiderBatch* expected_batch,
             checkOneScalarBatchEqual<double>(expected_child->as<ScalarBatch<double>>(),
                                              actual_child->as<ScalarBatch<double>>());
         break;
+      case SQLTypes::kDECIMAL:
+        // duckdb decimals use 16 bytes, but cider decimals use 8 bytes
```

Comment:
maybe in some cases we can compare ScalarBatch<int64_t/int32_t> and ScalarBatch<__int128_t>

Reply:
I don't think so, if "compare" refers to comparing with `memcmp`. For `int64_t`, its buffer uses 8 bytes for each value, for `int128_t`, the buffer uses 16 bytes. The buffer sizes are not aligned so a direct `memcmp` is not feasible. But running a for-loop and compare the data one by one is possible.